### ISSUE TITLE
[BUG]: Incorrect type for namespace parameter in login mutation. Should be Uint64, not Int.

### ISSUE THREAD

**Issue by suhabe** [dgraph-io/dgraph#8398]
### What version of Dgraph are you using?

v22.0.0

### Tell us a little more about your go-environment?

go version go1.19 darwin/arm64

### Have you tried reproducing the issue with the latest release?

Yes

### What is the hardware spec (RAM, CPU, OS)?

M1 MacOS, 64 GB

### What steps will reproduce the bug?

The `login` mutation in the admin schema uses Int instead of Uint64 for the namespace parameter:

``` 
type Mutation {
  ...

  """
  Login to Dgraph.  Successful login results in a JWT that can be used in future requests.
  If login is not successful an error is returned.
  """
  login(userId: String, password: String, namespace: Int, refreshToken: String): LoginPayload
}
```

**Comment by MichelDiz** [comment]
The mapping int->uint should be fine. Also, there's no uint support in GraphQL Spec.

Do you have anything to share? What problem did you have with that?

**Comment by suhabe** [comment]
Ok, we're confused because the admin schema already has a `scalar UInt64` type, but sometimes the admin schema uses its UInt64 scalar for namespace IDs and other times it uses Ints: https://dgraph.io/docs/graphql/admin/#admin

We code-generate a GraphQL client from the admin schema in Go, and thus the namespace ID sometimes typed as Go's `int` type and other times its typed as Go's `uint64` type. We're concerned that, due to the type-casting, the namespace ID changes in a way which would cause our system to log into the wrong tenant.

**Comment by MichelDiz** [comment]
Well, as far as I know all unsigned and signed int chase the same range. Only the signed that differs as to negative numbers(which cuts its range in half), but in the positive range they point to the same HEX. I think it would only be a problem for someone trying to apply negative numbers. But overall there's no problem with that mix.

You are right about the number of possible Namespaces. Because int64 divides the maximum 64-bit value by two. This makes it "only" 9 quintillion. 64 bits allows a range of 18 quintillion. Which is the case for uint64. But you'll never see a cluster with 100 million namespaces, not even imagine it with 9 quintillions.

>the namespace ID changes in a way which would cause our system to log into the wrong tenant

The worst that can happen is not being able to log in. If the number is out of range, it will not be possible to log in as the password does not match. If there is an extrapolation of the range (from 9 quintillion namespaces) you will soon notice, as you will not be able to log in. 

> What happens when the namespace ID is larger than the max limit of an int64?

Basically nothing. It will return an error and it will not be possible to create. But you will be a successful person if you manage to surpass the limit of 9 quintillion namespaces in production. :P

I think there could be some error with some IDE. If you notice anything wrong, we may need to intervene.

Last year the team wanted to reduce to uint32, as 64 is too big. 32 bit is much faster. But it would be a not very significant/important change. I think this must have been modified in version 21.12, but it has been rolled back.

**Comment by MichelDiz** [comment]
PS. Try this code https://go.dev/play/p/M4xo3NqcDyx

change `intNum := 3000` to `9200000000000000000` or any number. And you will see that they have the same number. But if you try negative numbers `intNum := -1` it will return `18446744073709551615` That is, it is returning the same range, but now in reverse. After 9 quintillion it stops. And you should use negative numbers. But that's not necessary. You won't have that scenario in a real situation.
